src/combinatorics/simple_graph/ends/defs.lean

```diff
@@ -0,0 +1,256 @@
+
+/-- Subsetship induces an obvious map on the induced graphs. -/
+@[reducible] def out_hom {K L} (h : K ⊆ L) : G.out L →g G.out K :=
+{ to_fun := λ v, ⟨v.val, set.compl_subset_compl.mpr h v.prop⟩,
+  map_rel' := by { rintros ⟨_, _⟩ ⟨_, _⟩, exact id } }
+
+lemma out_hom_refl (K) : G.out_hom (subset_refl K) = hom.id :=
+by { ext ⟨_, _⟩, refl, }
+
+lemma out_hom_trans {K L M} (h : K ⊆ L) (h' : L ⊆ M) :
+  G.out_hom (h.trans h') = (G.out_hom h).comp (G.out_hom h') :=
+by { ext ⟨_, _⟩, refl, }
+
+lemma out_hom_injective {K} {L} (h : K ⊆ L) : function.injective (G.out_hom h) :=
+by { rintros ⟨v, _⟩ ⟨w, _⟩ e,
+    simpa only [out_hom, subtype.mk_eq_mk, rel_hom.coe_fn_mk] using e, }
+
+end out
+
+/-- The components outside a given set of vertices `K` -/
+@[reducible] def comp_out := (G.out K).connected_component
+
+variables {G} {K L M}
+
+/-- The connected component of `v`. -/
+@[reducible] def comp_out_mk (G : simple_graph V) {v : V} (vK : v ∈ Kᶜ) : G.comp_out K :=
+connected_component_mk (G.out K) ⟨v, vK⟩
+
+/-- The set of vertices of `G` making up the connected component `C` -/
+@[reducible] def comp_out.supp (C : G.comp_out K) : set V :=
+{v : V | ∃ h : v ∈ Kᶜ, G.comp_out_mk h = C}
+
+@[ext] lemma comp_out.eq_iff_supp_eq (C D : G.comp_out K) : C = D ↔ C.supp = D.supp :=
+begin
+  split,
+  { rintro ⟨⟩, refl, },
+  { refine connected_component.ind₂ _ C D,
+    rintros ⟨v, hv⟩ ⟨w, hw⟩ h,
+    simp only [set.ext_iff, connected_component.eq, set.mem_compl_iff, set.mem_set_of_eq] at h ⊢,
+    exact ((h v).mp ⟨hv, reachable.refl _⟩).some_spec, }
+end
+
+instance : set_like (G.comp_out K) V :=
+{ coe := comp_out.supp,
+  coe_injective' := λ C D, (comp_out.eq_iff_supp_eq _ _).mpr, }
+
+@[simp] lemma comp_out.mem_supp_iff {v : V} {C : comp_out G K} :
+  v ∈ C ↔ ∃ (vK : v ∈ Kᶜ), G.comp_out_mk vK = C := iff.rfl
+
+lemma comp_out_mk_mem (G : simple_graph V) {v : V} (vK : v ∈ Kᶜ) :
+  v ∈ G.comp_out_mk vK := ⟨vK, rfl⟩
+
+lemma comp_out_mk_eq_of_adj (G : simple_graph V) {v w : V} (vK : v ∈ Kᶜ) (wK : w ∈ Kᶜ)
+  (a : G.adj v w) : G.comp_out_mk vK = G.comp_out_mk wK :=
+by { rw [connected_component.eq], apply adj.reachable, exact a }
+
+namespace comp_out
+
+/--
+A `comp_out` specialization of `quot.lift`, where soundness has to be proved only for adjacent
+vertices.
+-/
+def lift {β : Sort*} (f : ∀ ⦃v⦄ (hv : v ∈ Kᶜ), β)
+  (h : ∀ ⦃v w⦄ (hv : v ∈ Kᶜ) (hw : w ∈ Kᶜ) (a : G.adj v w), f hv = f hw) : G.comp_out K → β :=
+connected_component.lift (λ vv, f vv.prop) $ (λ v w p, by
+  { induction p with _ u v w a q ih,
+    { rintro _, refl, },
+    { rintro h', exact (h u.prop v.prop a).trans (ih h'.of_cons), } })
+
+lemma ind {β : G.comp_out K → Prop} (f : ∀ ⦃v⦄ (hv : v ∈ Kᶜ), β (G.comp_out_mk hv)) :
+  ∀ (C : G.comp_out K), β C := by
+{ apply connected_component.ind, exact λ ⟨v, vnK⟩, f vnK, }
+
+/-- The induced graph on the vertices `C`. -/
+@[reducible, protected]
+def coe_graph (C : comp_out G K) : simple_graph C.supp := G.induce (C : set V)
```

Comment:
This is a terrible name. We already have `simple_graph.subgraph.coe`, `simple_graph.subgraph.spanning_coe`, ... and this will add to the naming confusion.

Reply:
What would you propose?

Reply:
`induced_graph` ?

Reply:
I'd suggest `simple_graph.component_compl.coe_graph`, or you may delete it for now since it's not actually used ...
And why the type isn't `simple_graph C`? Isn't the set coercion supposed to be the simp normal form rather than `.supp`?

Reply:
OK, changed the type to `simple_graph C`. We can remove this def for now, but I use the coercion to graphs [here](https://github.com/bottine/mathlib/blob/39b0cc776009febe48df86465d2974f66a23e7ba/src/combinatorics/simple_graph/ends/properties2.lean#L177) so I'll need to bring it back eventually: we might as well settle for a name :)